app/code/core/Mage/Eav/Model/Entity/Collection/Abstract.php

```diff
@@ -1332,7 +1332,7 @@ protected function _addAttributeJoin($attributeCode, $joinType = 'inner')
         /**
          * process join type
          */
-        $joinMethod = $joinType === 'left' ? 'joinLeft' : 'join';
+        $joinMethod = $joinType === true || $joinType === 'left' ? 'joinLeft' : 'join';
```

Comment:
```suggestion
        $joinMethod = ($joinType === true || $joinType === 'left') ? 'joinLeft' : 'join';
```
dunno if that's needed

Reply:
It technically isn't because '||' binds "tighter" than '?:' so the boolean expression on the left is evaluated before the '?:'. On the other hand it could make things clearer in reading for some and doesn't hurt, does it?

Reply:
How about `in_array($joinType, [true, 'left'])` ?

Reply:
... or keep this as it is and add ...

```
if ($joinType === true) {
   $joinType = 'left';
}
```